docs/2-building-blocks-of-oop-part-1/hometask/src/Triangle.ts

```diff
@@ -1,3 +1,36 @@
-import { Shape } from './Shape';
+import { Shape } from './Shape'
+import { Point } from './Point';
 
-export class Triangle extends Shape {}
+export class Triangle extends Shape {
+  constructor(pointA: Point, pointB: Point, pointC: Point);
+  constructor(pointA: Point, pointB: Point, pointC: Point, color: string, filled: boolean);
+  constructor(pointA: Point, pointB: Point, pointC: Point, color?: string, filled?: boolean) {
+    if (typeof color === 'string' && typeof filled === 'boolean') {
+      super([pointA, pointB, pointC], color, filled);
+    } else {
+      super([pointA, pointB, pointC])
+    }
+  }
+
+  toString() {
+    let pointsStrArr: string[] = []
+    for (let i = 0; i < this.points.length; i++) {
+      pointsStrArr.push(`v${i + 1}=${this.points[i].toString()}`);
+    }
```

Comment:
Hi, if you would like to use to instead, and `toString` would be called automatically when using template string if I was not wrong.
```ts
this.points.map((point, i) => `v${i + 1}=${point}`
```